src/zap_hooks/helpers/auth_context.py

```diff
@@ -0,0 +1,371 @@
+
+    # add all found cookies as session cookies
+    for cookie in driver.get_cookies():
+        if zap is not None:
+            zap.httpsessions.set_session_token_value(
+                target, 'auth-session', cookie['name'], cookie['value'])
+        log(f"Cookie added: {cookie['name']}={cookie['value']}")
+
+    # add token from cookies if exists
+    add_token_from_cookie(zap, driver.get_cookies())
+
+    # Mark the session as active
+    if zap is not None:
+        zap.httpsessions.set_active_session(target, 'auth-session')
+        log(f"Active session: {zap.httpsessions.active_session(target)}")
+
+    # try to find JWT tokens in Local Storage and Session Storage and add them as Authorization header
+    localStorage = BrowserStorage(driver, 'localStorage')
+    sessionStorage = BrowserStorage(driver, 'sessionStorage')
+    add_token_from_browser_storage(zap, localStorage, config)
+    add_token_from_browser_storage(zap, sessionStorage, config)
+
+def validate_authentication_url(driver, url):
+    """Validate that the authentication url is called and returns a 200 status code"""
+    log(f"Validating authentication url: {url}")
+    url_found = False
+    for request in driver.requests:
+        if request.response and url in request.url:
+            url_found = True
+            log(f"Checking response status code {request.response}")
+            if request.response.status_code != 200:
+                log(f"Status code is not 200 for {request.url}, it is {request.response.status_code}")
+                sys.exit(1)
+            else: 
+                log(f"Status code is 200 for {request.url}, authentication was successful")
+    if not url_found:
+        log(f"Authentication url {url} was not found, authentication failed.")
+        sys.exit(1)
+
+def add_token_from_browser_storage(zap, browserStorage, config):
+    """Add JWT token from browser storage as Authorization header"""
+    for key in browserStorage:
+        log(f"Found key: {key}")
+        match = search('(eyJ[^"]*)', browserStorage.get(key))
+        if match:
+            auth_header = "Bearer " + match.group()
+            add_authorization_header(zap, auth_header)
+
+def add_token_from_cookie(zap, cookies):
+    """Add JWT token from cookies as Authorization header"""
+    for cookie in cookies:
+        if cookie['name'] == 'token':
+            auth_header = "Bearer " + cookie['value']
+            add_authorization_header(zap, auth_header)
+        
+
+def login_from_token_endpoint(zap, config):
+    """Login using a token endpoint"""
+    log('Fetching authentication token from endpoint')
+    response = post(config.auth_token_endpoint, data={
+        'username': config.auth_username, 'password': config.auth_password})
+    data = response.json()
+    auth_header = None
+
+    if "token" in data:
+        auth_header = f"Bearer {data['token']}"
+    elif "token_type" in data:
+        auth_header = f"{data['token_type']} {data['token_type']}"
```

Comment:
double check

Reply:
done

Reply:
what does the literal string look like for that case? `auth_header = f"{data['token_type']} {data['token_type']}"`

Reply:
fixed